### PR TITLE
Added A new Project (Churn Prediction)

### LINKED ISSUE #1686: ChurnScope Predictor
# Add ChurnScope — Telco Customer Churn Prediction Project

## Description

Add a machine learning project named **ChurnScope** that predicts telecom customer churn using the IBM Watson Telco Customer Churn dataset.

The project focuses on customer behavior analysis, feature engineering, and churn prediction using classical machine learning models.

## Features

* Customer churn prediction system
* Data preprocessing and cleaning
* Feature engineering pipeline
* Exploratory Data Analysis (EDA)
* Multiple ML model comparison
* ROC-AUC and recall-based evaluation
* Gradient Boosting based final prediction model

## Project Structure

```bash
ChurnScope/
├── data/
│   └── Telco-Customer-Churn.csv
├── notebook/
│   └── Churn_Prediction.ipynb
└── README.md
```

## Models Used

* Logistic Regression
* Random Forest
* Gradient Boosting
* XGBoost

## Expected Outcome

* Predict customer churn probability
* Identify important churn-driving factors
* Compare classical ML algorithms for churn prediction
* Provide a structured ML workflow for learning and experimentation

## Additional Notes

Gradient Boosting achieved the best ROC-AUC performance among the evaluated models and was selected as the final model for churn prediction.

### PR DESCRIPTION
**Pull request by kanan010506**
# Description

Added  **ChurnScope — Telco Customer Churn Prediction** project.

## About the Project

ChurnScope is a machine learning project focused on predicting telecom customer churn using the IBM Watson Telco Customer Churn dataset. The project performs customer behavior analysis and identifies users who are likely to discontinue telecom services.

The folder includes:

* Dataset used for churn analysis
* Jupyter notebook containing:

  * Data preprocessing
  * Feature engineering
  * Exploratory analysis
  * Model training
  * Model evaluation
* Comparative analysis of multiple ML algorithms
* Final churn prediction pipeline using Gradient Boosting

## Key Features

* Customer churn prediction using classical ML models
* End-to-end preprocessing pipeline with scikit-learn
* Feature engineering for improved model performance
* ROC-AUC and recall-based model evaluation
* Analysis of important churn-driving factors

## Models Evaluated

* Logistic Regression
* Random Forest
* Gradient Boosting
* XGBoost

Gradient Boosting achieved the best overall ROC-AUC performance and was selected as the final model.

## Type of Change

* [x] Documentation update
* [x] Bug fix
* [ ] New feature
* [x] Breaking change

## Checklist

* [] README is properly formatted
* [] Added detailed project explanation
* [] Improved repository documentation
* [] No existing functionality affected

## Related Issue
fixes #1686 
Closes #1686

### CODE REVIEW

**Comment by kanan010506** [comment]
This is Under Gssoc'26

File: ChurnScope/data/Telco-Customer-Churn.csv
Comment (Sibam-Paul) [inline ChurnScope/data/Telco-Customer-Churn.csv:1]: remove

File: ChurnScope/README.md
Comment (Sibam-Paul) [inline ChurnScope/README.md:1]: follow this format of readme #1952

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
@kanan010506 everything looks good
just make this small change
1. remove the csv file and attach it using `gdown` then re-run the cells and then commit
2. follow the readme format #1952 make it look detailed

**Comment by Sibam-Paul** [comment]
@kanan010506 make the changes please

**Comment by kanan010506** [comment]
@Sibam-Paul  pls check

**Review by Sibam-Paul — APPROVED** [review]
Looks good

**Comment by kanan010506** [comment]
@Sibam-Paul what's the issue ?

**Comment by Sibam-Paul** [comment]
@kanan010506 thank you for ur contribution :)
